### ISSUE TITLE
Crash when navigating in did-fail-load event

### ISSUE THREAD

**Issue by cumajkeee** [electron/electron#8700]
I have a download link (40mb+ file) when clicked and confirm where to save file, make another click on same link and got crash. 

Tried to make sample of crash, and didn't yet face such behavior: on second click app didn't made something (not showed download dialog), but also not crashed. Will provide sample as soon as got reproducible one.

Debug:
<img width="1331" alt="screen shot 2017-02-16 at 1 24 09 pm" src="https://cloud.githubusercontent.com/assets/1834053/23017440/56041e32-f44b-11e6-8d5c-f1ff6ad671cd.png">

Report:
<img width="1398" alt="screen shot 2017-02-16 at 1 25 28 pm" src="https://cloud.githubusercontent.com/assets/1834053/23017568/db45549e-f44b-11e6-8f74-650b0f2cc756.png">

Any idea what can affect such problem? 

As I see from report last triggered method was webContents.loadURL, but didn't understand why it was called since I have webContetst.session event 'will-download' called, that had to be triggered after webContents.downloadURL, not loadURL.

**Comment by cumajkeee** [comment]
[download-crash.zip](https://github.com/electron/electron/files/780128/download-crash.zip)

Reproduce sample, seems some how 'did-fail-load' had affect on it.

**Comment by kevinsawicki** [comment]
@cumajkeee can you try your sample using `electron@1.6.0` and see if you still see the crash?

I'm not having success reproducing it using that version, each time I click the link I get the save dialog and don't see any crashes.

**Comment by cumajkeee** [comment]
@kevinsawicki yes, it reproduce in 1.6.0 also. Did you start download when clicked first time?

**Comment by deepak1556** [comment]
I am able to reproduce with the provided test case, have reduced the sample to

```
const {app, BrowserWindow} = require('electron')

let mainWindow

app.on('ready', () => {
  mainWindow = new BrowserWindow()

  const source = `
    <body>
      // some unreachable url.
      <a href="https://electron.io">Link</a>
    </body>
    <script>
      setInterval(function () {
        document.getElementsByTagName('a')[0].click()
      }, 100)
    </script>
  `

  mainWindow.webContents.on('did-fail-load', () => {
    mainWindow.loadURL('about:blank')
  })

  mainWindow.loadURL(`data:text/html,${source}`)
})

app.on('quit', () => {
  app.quit()
})
```

```
[21764:0217/235105.321064:FATAL:navigation_controller_impl.cc(2021)] Check failed: !in_navigate_to_pending_entry_ || delegate_->IsBeingDestroyed(). 
```

**Comment by kevinsawicki** [comment]
Reopening this, the original fix has been reverted in #9674 since it introduced other issues.

**Comment by codebytere** [comment]
We are no longer implementing bugfixes for versions of Electron <= `1.7.x`, so i'm going to close this issue but if it is still persisting in more recent versions of Electron we can certainly reopen it!
